eclair-core/src/test/scala/fr/acinq/eclair/payment/send/OfferPaymentSpec.scala

```diff
@@ -0,0 +1,134 @@
+    val offerPayment = testKit.spawn(OfferPayment(nodeParams, postman.ref, paymentInitiator.ref))
+    val routeParams = nodeParams.routerConf.pathFindingExperimentConf.getRandomConf().getDefaultRouteParams
+    try {
+      withFixture(test.toNoArgTest(FixtureParam(offerPayment, nodeParams, postman, paymentInitiator, routeParams)))
+    } finally {
+      testKit.stop(offerPayment)
+    }
+  }
+
+  implicit val classicSystem: ActorSystem = system.classicSystem
+
+  test("basic offer payment") { f =>
+    import f._
+
+    val probe = TestProbe()
+    val merchantKey = randomKey()
+
+    val offer = Offer(None, "amountless offer", merchantKey.publicKey, Features.empty, nodeParams.chainHash)
+    offerPayment ! PayOffer(probe.ref, offer, 40_000_000 msat, 1, SendPaymentConfig(None, 1, routeParams, blocking = false))
+    val Postman.SendMessage(_, Recipient(recipientId, _, _), _, message, replyTo, _) = postman.expectMessageType[Postman.SendMessage]
+    assert(recipientId == merchantKey.publicKey)
+    assert(message.get[OnionMessagePayloadTlv.InvoiceRequest].nonEmpty)
+    val Right(invoiceRequest) = InvoiceRequest.validate(message.get[OnionMessagePayloadTlv.InvoiceRequest].get.tlvs)
+
+    val preimage = randomBytes32()
+    val paymentRoute = PaymentBlindedRoute(RouteBlinding.create(randomKey(), Seq(merchantKey.publicKey), Seq(hex"7777")).route, PaymentInfo(0 msat, 0, CltvExpiryDelta(0), 0 msat, 1_000_000_000 msat, Features.empty))
+    val invoice = Bolt12Invoice(invoiceRequest, preimage, merchantKey, 1 minute, Features.empty, Seq(paymentRoute))
+    replyTo ! Postman.Response(FinalPayload(TlvStream(OnionMessagePayloadTlv.Invoice(invoice.records)), TlvStream.empty))
+    val send = paymentInitiator.expectMsgType[SendPaymentToNode]
+    assert(send.invoice == invoice)
+
+    val paymentId = UUID.randomUUID()
+    send.replyTo ! paymentId
+    val uuid = probe.expectMsgType[UUID]
+    assert(uuid == paymentId)
+
+    TypedProbe().expectTerminated(offerPayment)
+  }
+
+  test("no reply to invoice request with retries") { f =>
+    import f._
+
+    val probe = TestProbe()
+    val merchantKey = randomKey()
+
+    val offer = Offer(None, "amountless offer", merchantKey.publicKey, Features.empty, nodeParams.chainHash)
+    offerPayment ! PayOffer(probe.ref, offer, 40_000_000 msat, 1, SendPaymentConfig(None, 1, routeParams, blocking = false))
+    for (i <- 1 to nodeParams.onionMessageConfig.maxAttempts) {
+      val Postman.SendMessage(_, Recipient(recipientId, _, _), _, message, replyTo, _) = postman.expectMessageType[Postman.SendMessage]
+      assert(recipientId == merchantKey.publicKey)
+      assert(message.get[OnionMessagePayloadTlv.InvoiceRequest].nonEmpty)
+      val Right(invoiceRequest) = InvoiceRequest.validate(message.get[OnionMessagePayloadTlv.InvoiceRequest].get.tlvs)
+      assert(invoiceRequest.isValidFor(offer))
+      replyTo ! Postman.NoReply
+    }
+    probe.expectMsg(NoInvoice)
+    paymentInitiator.expectNoMessage()
```

Comment:
Always specify a duration in `expectNoMessage`, otherwise the default seems to be 500ms which can quickly aggregate and slow down the test suite for no good reason since we have thousands of tests...

```suggestion
    paymentInitiator.expectNoMessage(50 millis)
```